src/transformers/trainer.py

```diff
@@ -609,6 +611,16 @@ def create_optimizer_and_scheduler(self, num_training_steps: int):
         """
         Setup the optimizer and the learning rate scheduler.
 
+        We provide a reasonable default that works well. If you want to use something else, you can pass a tuple in the
+        Trainer's init through :obj:`optimizers`, or subclass and override this method in a subclass.
+        """
+        self.create_optimizer()
+        self.create_scheduler(num_training_steps)
+
+    def create_optimizer(self):
```

Comment:
@sgugger, if you could validate that it's OK how @cli99 split the optimizer/scheduler creation